### PR TITLE
use fp8 sdp in llama

### PR DESCRIPTION
**Pull request by MeouSker77**
## Description

use fp8 sdp in llama with transformers 4.31.

Note: current fp8 sdp and fp16 sdp both don't support attention_mask, we assume masks are all zeros

### CODE REVIEW

**Review by rnwang04 — APPROVED** [review]
LGTM
